### PR TITLE
add mod tidy hack to all the service packages

### PR DESCRIPTION
**Pull request by ArcturusZhang**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
I have requested those "invalid" links, they should be valid very soon

File: sdk/network/armnetwork/go_mod_tidy_hack.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// +build modhack
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+// This file, and the github.com/Azure/azure-sdk-for-go import, won't actually become part of
+// the resultant binary.
+
+package armcompute
```
Comment (jhendrixMSFT) [inline sdk/network/armnetwork/go_mod_tidy_hack.go:9]: armnetwork

File: sdk/resources/armresources/go_mod_tidy_hack.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// +build modhack
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+// This file, and the github.com/Azure/azure-sdk-for-go import, won't actually become part of
+// the resultant binary.
+
+package armcompute
```
Comment (jhendrixMSFT) [inline sdk/resources/armresources/go_mod_tidy_hack.go:9]: armresources

File: sdk/storage/armstorage/go_mod_tidy_hack.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// +build modhack
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+// This file, and the github.com/Azure/azure-sdk-for-go import, won't actually become part of
+// the resultant binary.
+
+package armcompute
```
Comment (jhendrixMSFT) [inline sdk/storage/armstorage/go_mod_tidy_hack.go:9]: armstorage

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

File: sdk/network/armnetwork/go_mod_tidy_hack.go (hunk 1 above)
Comment (ArcturusZhang) [reply sdk/network/armnetwork/go_mod_tidy_hack.go:9]: Oops i forgot to change it when i copied it

**Comment by ArcturusZhang** [comment]
Hi @jhendrixMSFT I just fixed the package names

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**